mapie/time_series_regression.py

```diff
@@ -0,0 +1,325 @@
+                ).T  # type: ignore
+                higher_quantiles = masked_quantile(
+                    ma.masked_invalid(self.conformity_scores_),
+                    1 - alpha_np + betas_0,
+                    axis=0,
+                    method="higher",
+                ).T  # type: ignore
+
+                if (self.method in ["naive", "base"]) or (self.cv == "prefit"):
+                    y_pred_low = np.column_stack(
+                        [
+                            y_pred[:, np.newaxis] + lower_quantiles[k]
+                            for k in range(len(alpha_np))
+                        ]
+                    )
+                    y_pred_up = np.column_stack(
+                        [
+                            y_pred[:, np.newaxis] + higher_quantiles[k]
+                            for k in range(len(alpha_np))
+                        ]
+                    )
+                else:
+                    y_pred_multi = np.column_stack(
+                        [e.predict(X) for e in self.estimators_]
+                    )
+
+                    # At this point, y_pred_multi is of shape
+                    # (n_samples_test, n_estimators_). The method
+                    # ``aggregate_with_mask`` fits it to the right size thanks
+                    # to the shape of k_.
+
+                    y_pred_multi = self.aggregate_with_mask(
+                        y_pred_multi, self.k_
+                    )
+
+                    if self.method == "plus":
+                        pred = aggregate_all(self.agg_function, y_pred_multi)
+                        y_pred_low = np.column_stack(
+                            [
+                                pred + lower_quantiles[k]
+                                for k in range(len(alpha_np))
+                            ]
+                        )
+                        y_pred_up = np.column_stack(
+                            [
+                                pred + higher_quantiles[k]
+                                for k in range(len(alpha_np))
+                            ]
+                        )
+
+                    if self.method == "minmax":
+                        lower_bounds = np.min(
+                            y_pred_multi, axis=1, keepdims=True
+                        )
+                        upper_bounds = np.max(
+                            y_pred_multi, axis=1, keepdims=True
+                        )
+                        y_pred_low = np.column_stack(
+                            [
+                                lower_bounds + lower_quantiles[k]
+                                for k in range(len(alpha_np))
+                            ]
+                        )
+                        y_pred_up = np.column_stack(
+                            [
+                                upper_bounds + higher_quantiles[k]
+                                for k in range(len(alpha_np))
+                            ]
+                        )
```

Comment:
This part is very similar to the one of `MapieRegressor` (except for the residuals are no longer symmetric). Would it be worth creating a dedicated method in `MapieRegressor` that can be used also in `MapieTimeSeriesRegressor` ?

Reply:
In a dedicated PR?